### PR TITLE
Feature/CVSB-16162 - Improvement of 10278 PUT applicable fields logic 

### PR DESCRIPTION
**Pull request by irina005**
Ticket:
https://jira.dvsacloud.uk/browse/CVSB-16162

This PR includes also a refactoring of the test-record.mapper (changed and extracted the applicable test type objects into enums).

### CODE REVIEW

File: src/app/test-record/test-record.mapper.ts [hunk 1]
```diff
@@ -212,7 +112,8 @@ export class TestRecordMapper {
     testTypeMapped.emissionStandard = testResultFormData.testType.emissionStandard;
     testTypeMapped.smokeTestKLimitApplied = testResultFormData.testType.smokeTestKLimitApplied;
     testTypeMapped.fuelType = testResultFormData.testType.fuelType;
-    if (!!testTypeMapped.modType) {
+
+    if (!!testTypeMapped.modType && testResultMapped.vehicleType === 'trl') {
```
Comment (JonLazariniDD) [inline src/app/test-record/test-record.mapper.ts:116]: Let's please log a task to cover that later as the mapper unit tests seem to be covering not all the permutations/logic we have.
We will cover that later if we have to ship it now please.

File: src/app/test-record/test-record.mapper.ts [hunk 2]
```diff
@@ -251,45 +145,60 @@ export class TestRecordMapper {
...
-
-    if (!testTypesApplicable.testSectionApplicable1.hasOwnProperty(testTypeMapped.testTypeId)) {
+    if (!testTypesApplicable.anniversaryDateApplicable[testTypeMapped.testTypeId]) {
       testTypeMapped.certificateNumber = undefined;
       testTypeMapped.testExpiryDate = undefined;
       testTypeMapped.testAnniversaryDate = undefined;
     }
 
-    if (!testTypesApplicable.testSectionApplicable2.hasOwnProperty(testTypeMapped.testTypeId)) {
+    if (
```
Comment (JonLazariniDD) [inline src/app/test-record/test-record.mapper.ts:154]: Same for this please. Could in the same ticket.

File: src/app/test-record/vehicle/edit/vehicle-edit.component.html [hunk 3]
```diff
@@ -80,39 +81,41 @@
...
+                  Odometer reading unit
+                </label>
+                <select
+                  class="govuk-select"
+                  id="test-odometerReadingUnit"
+                  formControlName="odometerReadingUnits"
                 >
-              </select>
+                  <option
+                    *ngFor="let unit of odometerReadingUnits"
```
Comment (JonLazariniDD) [inline src/app/test-record/vehicle/edit/vehicle-edit.component.html:110]: Do we need ids for those? We can check with automation testers.

File: src/app/test-record/test-record.mapper.ts (hunk 1 above)
Comment (irina005) [reply src/app/test-record/test-record.mapper.ts:116]: https://jira.dvsacloud.uk/browse/CVSB-16228

File: src/app/test-record/test-record.mapper.ts (hunk 2 above)
Comment (irina005) [reply src/app/test-record/test-record.mapper.ts:154]: https://jira.dvsacloud.uk/browse/CVSB-16228

File: src/app/test-record/defects/defects.component.spec.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { DefectsComponent } from './defects.component';
 import { SharedModule } from '@app/shared/shared.module';
-import { TEST_MODEL_UTILS } from '@app/utils/test-models.utils';
 import { Defect } from '@app/models/defect';
+import { TESTING_TEST_MODELS_UTILS } from '@app/utils/testing-test-models.utils';
```
Comment (alleypa) [inline src/app/test-record/defects/defects.component.spec.ts:5]: please use `TEST_MODEL_UTILS` for your export. Filename to just be `test-model.utils`

**Pull request merged**